pg_search/src/postgres/options.rs

```diff
@@ -125,6 +126,16 @@ extern "C" fn validate_uuid(value: *const std::os::raw::c_char) {
     cstr_to_rust_str(value);
 }
 
+#[pg_guard]
+extern "C" fn validate_memory_budget(value: *const std::os::raw::c_char) {
+    let memory_budget_str = cstr_to_rust_str(value);
+    if !memory_budget_str.is_empty() {
+        memory_budget_str
+            .parse::<i32>()
+            .expect("Invalid memory budget value");
+    }
+}
+
```

Comment:
If the user is going to see this error message, we should try and make it as helpful as possible so they can be aware of the valid range of values.